lib/timetable-form/Form.tsx

```diff
@@ -40,18 +61,28 @@ export const Form = (props: React.HTMLAttributes<HTMLDivElement>) => {
       <div className={styles.formRow}>
         <TextInput
           label="Local planning authority"
-          onChange={setLpa}
-          value={lpa}
+          onChange={(lpa) =>
+            setDevelopmentPlan((prev) => ({
+              ...prev,
+              organisations: [lpa],
+            }))
+          }
+          value={developmentPlan.organisations[0]}
         />
       </div>
-      {stageNames.map((stageName) => (
-        <div key={stageName}>
+      {developmentPlan.timetableEvents.map((stage) => (
+        <div key={stage.developmentPlanEvent}>
           <DateInput
-            value={stages[stageName]}
-            label={stageName}
-            name={`${stageName.split(" ").join("-")}-date`}
+            value={stage.eventDate}
+            label={stage.developmentPlanEvent}
+            name={`${stage.developmentPlanEvent.split(" ").join("-")}-date`}
             onChange={(value) =>
-              setStages((prev) => ({ ...prev, [stageName]: value }))
+              setDevelopmentPlan((prev) => ({
+                ...prev,
+                timetableEvents: prev.timetableEvents.map((e) =>
+                  e === stage ? { ...e, eventDate: value } : e
+                ),
```

Comment:
Looks like what we want here is a `.find` rather than a `.map`? Only one element is being updated so we don't need to loop through them all

Reply:
applied the change. Thanks Ronan :)